### PR TITLE
Added FlipHorizontal and FlipVertical

### PR DESCRIPTION
**Pull request by strubelz**
Updated version of this: http://forum.adamslair.net/viewtopic.php?f=4&t=433

### CODE REVIEW

**Comment by ilexp** [comment]
Very focused and small patch - good choice for a first PR! However, there are some problems:

- Mixed up spaces and tabs. Duality uses [tabs for indentation](https://github.com/AdamsLair/duality/wiki/How-to-Contribute#step-three-develop-the-framework). Spaces are only used for non-indent alignment.
- The code doesn't build, see the [AppVeyor build details](https://ci.appveyor.com/project/AdamsLairBot/duality/build/AppVeyor%20master%20Build%2047/messages) for more information.

Please fix these problems, so your PR can be reconsidered for a merge.

**Comment by strubelz** [comment]
For some reason my Visual Studio replaces tabs with spaces. Is there a workaround for this?

**Comment by ilexp** [comment]
> For some reason my Visual Studio replaces tabs with spaces. Is there a workaround for this?

Your Visual Studio settings allow to configure the preferred indentation mark on a per-language basis. I'm not sure whether this is also possible on a per-project or per-solution basis though.

File: Duality/Components/Renderers/SpriteRenderer.cs [hunk 1]
```diff
@@ -179,38 +198,43 @@ protected void PrepareVertices(ref VertexC1P3T2[] vertices, IDrawDevice device,
...
 			vertices[3].Pos.Y = posTemp.Y + edge4.Y;
 			vertices[3].Pos.Z = posTemp.Z + this.VertexZOffset;
-			vertices[3].TexCoord.X = uvRect.RightX;
-			vertices[3].TexCoord.Y = uvRect.Y;
-			vertices[3].Color = mainClr;
-			
-			if (this.pixelGrid)
+			vertices[3].TexCoord.X = right;
+			vertices[3].TexCoord.Y = top;
+            vertices[3].Color = mainClr;
```
Comment (ilexp) [inline Duality/Components/Renderers/SpriteRenderer.cs]: Leftover space indentation :) Also, line 237, just below.

**Comment by ilexp** [comment]
Compiles just fine now. Will consider this PR for a future merge! I might go ahead and replace the two flip-booleans with a bitmask enum afterwards. Not sure yet though.

**Comment by ilexp** [comment]
Merged. For details what changed, see [here](https://github.com/AdamsLair/duality/commit/f1818b72dcca43be6f37624ab2436b650d3f302d).

**Pull request merged**